src/sentry/db/postgres/transactions.py

```diff
@@ -33,10 +33,11 @@ def django_test_transaction_water_mark(using: str | None = None):
         return
 
     from sentry.testutils import hybrid_cloud
-    from sentry.testutils.silo import assume_test_silo_mode
 
     # Exempt get_connection call from silo validation checks
-    with assume_test_silo_mode(SiloMode.MONOLITH):
+    with SiloMode.exit_single_process_silo_context(), SiloMode.enter_single_process_silo_context(
```

Comment:
This also addresses an issue where the `django_test_transaction_water_mark` would not work inside of an RPC call due to the way... override_settings works.  Fortunately, the `SiloMode.enter_single_process_silo_context` addresses this.